src/clientiface.cpp

```diff
@@ -172,20 +179,25 @@ void RemoteClient::GetNextBlocks (
 	// get view range and camera fov from the client
 	s16 wanted_range = sao->getWantedRange();
 	float camera_fov = sao->getFov();
-	// if FOV, wanted_range are not available (old client), fall back to old default
-	if (wanted_range <= 0) wanted_range = 1000;
-	if (camera_fov <= 0) camera_fov = (72.0*M_PI/180) * 4./3.;
 
-	const s16 full_d_max = MYMIN(g_settings->getS16("max_block_send_distance"), wanted_range);
-	const s16 d_opt = MYMIN(g_settings->getS16("block_send_optimize_distance"), wanted_range);
+	// When the player moves and looks forward, try to send some extra blocks if possible.
+	wanted_range += adjust / MAP_BLOCKSIZE;
+
+	const s16 full_d_max = std::min(g_settings->getS16("max_block_send_distance"), wanted_range);
+	const s16 d_opt = std::min(g_settings->getS16("block_send_optimize_distance"), wanted_range);
 	const s16 d_blocks_in_sight = full_d_max * BS * MAP_BLOCKSIZE;
-	//infostream << "Fov from client " << camera_fov << " full_d_max " << full_d_max << std::endl;
 
 	s16 d_max = full_d_max;
-	s16 d_max_gen = MYMIN(g_settings->getS16("max_block_generate_distance"), wanted_range);
+	s16 d_max_gen = std::min(g_settings->getS16("max_block_generate_distance"), wanted_range);
+
+	// Reduce the field of view when a player moves and looks forward.
+	// Why 50? It works well with speeds between 0 and 100 (adjusted for BS).
+	camera_fov = camera_fov / (1 + adjust / 50.0f);
```

Comment:
Reducing the FOV allows the server to send fewer blocks.
If the player moves _and_ looks in the direction of movements the blocks on the side are instantly out of view anyway.

At very fast movements (ethereal with crystal books on :) ) the effect is visible on the sides, but the server now has a fighting chance to keep with the blocks that the player is looking at.